### ISSUE TITLE
[BUG] TypeError: Cannot read property 'text' of undefined

### ISSUE THREAD

**Issue by rabraghib** [compodoc/compodoc#1041]
Failed generating documentation.
```
Unhandled Rejection at: Promise {
  <rejected> TypeError: Cannot read property 'text' of undefined
      at ClassHelper.visitTypeName (...\node_modules\@compodoc\compodoc\dist\application54cd2170.js:6467:22)
      at ClassHelper.visitTypeName (...\node_modules\@compodoc\compodoc\dist\application54cd2170.js:6470:21)
      at ClassHelper.visitType (...\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:6485:32)
      at parseNode (...\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:8395:56)
      at ...\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:8575:13
      at visitNodes (...\node_modules\ts-simple-ast\node_modules\typescript\lib\typescript.js:14786:30)
      at Object.forEachChild (...\node_modules\ts-simpleast\node_modules\typescript\lib\typescript.js:15012:24)
      at AngularDependencies.getSourceFileDecorators (...\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:8155:16)
      at ...\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:7960:31
      at Array.map (<anonymous>)} reason: TypeError: Cannot read property 'text' of undefined
    at ClassHelper.visitTypeName (...\node_modules\@compodoc\compodoc\dist\application54cd2170.js:6467:22)
    at ClassHelper.visitTypeName (...\node_modules\@compodoc\compodoc\dist\application54cd2170.js:6470:21)
    at ClassHelper.visitType (...\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:6485:32)
    at parseNode (...\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:8395:56)
    at ...\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:8575:13
    at visitNodes (...\node_modules\ts-simple-ast\node_modules\typescript\lib\typescript.js:14786:30)
    at Object.forEachChild (...\node_modules\ts-simple-ast\node_modules\typescript\lib\typescript.js:15012:24)
    at AngularDependencies.getSourceFileDecorators (...\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:8155:16)
    at ...\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:7960:31
    at Array.map (<anonymous>)
   ```
I am facing this error only when I declare a custom typescript type.
```typescript
export type entityName = "shortName1" | "shortName2" | "shortName3"; // ...
export type IRI = `/${ entityName }s${ "" | `/${   string }`}`;
```

**Comment by rabraghib** [comment]
@compodoc

**Comment by berkayyildiz** [comment]
I get exact same issue with this type of code

`type ServiceCode = `${Service}: ${ServiceReason}`
type ComponentCode = `${Component}: ${ComponentReason}`
type GeneralCode = `${Origin}: ${GeneralReason}`
`

**Comment by vogloblinsky** [comment]
fixed by c7cd839 but need to be completely implemented with https://github.com/compodoc/compodoc/issues/1053
